packages/json-spell-checker/index.js

```diff
@@ -0,0 +1,402 @@
+            path = 'r' + path
+        }
+
+        findPathsInSchema(getRefObjectOfSchema(SCHEMA, schema[REF]), path, references + `${schema[REF]},`, lookupTable)
+    }
+
+    if (schema.type === 'object') {
+        const properties = schema[PROPERTIES]
+        for (const key in properties) {
+            if (properties.hasOwnProperty(key)) {
+                if (key === REF) {
+                    
+                    if (path[0] === 'r' && references.includes(properties[key])) {
+                        continue
+                    }
+
+                    if (references.includes(properties[key])) {
+                        path = 'r' + path
+                    }
+
+                    findPathsInSchema(getRefObjectOfSchema(SCHEMA, properties[key]), path, references + `${properties[key]},`, lookupTable)
+                } else {
+                    findPathsInSchema(properties[key], `${path}${key}/`, references, lookupTable)
+                    keys.add(key)
+                }
+            }
+        }
+
+        const additionalProperties = schema[ADDITONAL_PROPERTIES]
+        for (const key in additionalProperties) {
+            if (additionalProperties.hasOwnProperty(key)) {
+                if (key === REF) {
+                    if (path[0] === 'r' && references.includes(additionalProperties[key])) {
+                        continue
+                    }
+
+                    if (references.includes(additionalProperties[key])) {
+                        path = 'r' + path
+                    }
+
+                    findPathsInSchema(getRefObjectOfSchema(SCHEMA, additionalProperties[key]), `${path}*/`, references + `${additionalProperties[key]},`, lookupTable)
+                }
+            }
+        }
+
+        const patternProperties = schema[PATTERN_PROPERTIES]
+        for (const key in patternProperties) {
+            if (patternProperties.hasOwnProperty(key)) {
+                if (key === REF) {
+                    if (path[0] === 'r' && references.includes(patternProperties[key])) {
+                        continue
+                    }
+
+                    if (references.includes(patternProperties[key])) {
+                        path = 'r' + path
+                    }
+
+                    findPathsInSchema(getRefObjectOfSchema(SCHEMA, patternProperties[key]), `${path}*/`, references + `${patternProperties[key]},`, lookupTable)
+                } else {
+                    findPathsInSchema(patternProperties[key], `${path}*/`, references, lookupTable)
+                }
+            }
+        }
+
+        putKeysToPath(lookupTable, path, keys)
+    }
+
+    if (schema.type === 'array') {
+        const items = schema.items
+
+        for (const item in items) {
+            if (items.hasOwnProperty(item)) {
+                if (item === REF) {
+                    if (path[0] === 'r' && references.includes(items[item])) {
+                        continue
+                    }
+
```

Comment:
please add comments on why r is added

Reply:
Fixed at [a9f2bdd](https://github.com/thingweb/thingweb-playground/pull/443/commits/a9f2bdd96a2a0fc3404919c280cec59291109d2c)